src/pyscaffold/cli_parser.py

```diff
@@ -0,0 +1,182 @@
+        return group
+
+    def _get_prompt_actions(self, extensions: List[Extension]) -> List[Action]:
+        prompt_actions = []
+        for action in self.actions:
+            if action in self.noprompts:
+                continue
+            if action.dest == "help":
+                continue
+            if action in self.called_actions:
+                continue
+            # check if action was included by another, e.g. cirrus -> pre-commit
+            if is_included(action, extensions):
+                continue
+            prompt_actions.append(action)
+
+        return prompt_actions
+
+    def format_flag(self, action: Action) -> str:
+        flag = action.option_strings[-1]
+        formatter = self._get_formatter()
+        args = formatter._format_args(action, action.dest)
+        return f"{flag} {args}"
+
+    def prompt_user(self, opts: ScaffoldOpts) -> ScaffoldOpts:
+        """Prompt the user for additional options
+
+        ToDo: Give a nice error message if user cancels the prompting with ctrl+c
+        ToDo: Show the user the defaults from `actions.get_default_options` by
+         extracting this functionality into smaller functions that can be applied here
+        """
+        argument_msg = "Arguments of [{flag}] (enter for default)"
+        activate_msg = "Activate"
+        choices_msg = "Choices"
+
+        for action in self._get_prompt_actions(opts["extensions"]):
+            print(opts["extensions"])
+            if is_included(action, opts["extensions"]):
+                continue
+            flag = self.format_flag(action)
+            print("Flag:", flag)
+            print("Help:", action.help)
+
+            if action.nargs == 0:
+                input = inquirer.confirm(activate_msg, default=False)
+            elif action.nargs == "?":
+                input = inquirer.confirm(activate_msg, default=False)
+                if input:
+                    input = inquirer.text(
+                        argument_msg.format(flag=flag), default=action.default
+                    )
+                else:
+                    input = None
+            elif action.choices:
+                input = inquirer.list_input(choices_msg, choices=action.choices)
+            else:
+                input = inquirer.text(argument_msg.format(flag=flag), default=None)
+            opts = merge_user_input(opts, input, action)
+
+        return opts
+
+
+def is_included(action: Action, extensions: List[Extension]):
+    """Checks if action was included by another extension
+
+    An extension can include another using :obj:`~pyscaffold.extensions.include`
+
+    FIXME: This still doesn't work!
+    """
+    extension = getattr(action, "const", None)
+    if hasattr(extension, "name"):
+        return extension.name in [ext.name for ext in extensions]
+    else:
+        return False
+
+
+def merge_user_input(opts: ScaffoldOpts, input: Any, action: Action) -> ScaffoldOpts:
```

Comment:
I suppose we need to convert the opts back to a namespace object, pipe it to the Action `__call__` and then convert it back to a dict... Should we try to do this before converting the namespace object into a dictionary to avoid going back and forth?

Reply:
I have always found the namespace object of `argparse` quite annoying as it creeps everywhere into your code if you are not careful and it feels not as natural as a dictionary, which is expected by many other libraries. So my first intuition would be to draw the line at the interface between argparse and Pyscaffold where the conversion should take place. So PyScaffold only deals with dictionaries everywhere and only when `argparse` functionality is called, it's converted to a namespace object. Or would you have an elegant solution to keep handling the namespace object to a minimum, so that it doesn't creep everywhere into the codebase?

Reply:
Ok, that makes sense, and we don't have to be super efficient anyway...

I suppose `merge_user_input`, would be something along the lines(?) :

```python
def merge_user_input(parser: ArgumentParser, opts: ScaffoldOpts, input: Any, action: Action) -> ScaffoldOpts:
      namespace = argparse.Namespace(**opts)
      action(parser, namespace, input, action.option_strings[-1])
      return vars(namespace)
```

The action should mutate the namespace as a side effect, so I suppose this will automatically merge the existing opts with any augmentation coming from the action...